### ISSUE TITLE
Silent wrong gradient: eager (non-jit) VJP of x[:, idx, :] gather zeroes batch rows on TPU (regression in libtpu 0.0.42)

### ISSUE THREAD

**Issue by m-braganca** [jax-ml/jax#39025]
### Description

On TPU, computing the gradient of a simple array-indexing gather with no jitting returns silently incorrect values. The same computation under jit is correct. This is a regression: it appears on jax 0.10.2 / libtpu 0.0.42.1 and is not present on jax 0.10.1 / libtpu 0.0.41.

### Reproduction

```python
import jax
import jax.numpy as jnp
import numpy as np

B, D, K = 256, 8, 128
xi = jnp.arange(D)                 # identity gather over the middle axis
gather = lambda x: x[:, xi, :]

x = jax.random.normal(jax.random.key(0), (B, D, K))
g = jax.random.normal(jax.random.key(1), (B, D, K))

dx_no_jit = np.asarray(jax.vjp(gather, x)[1](g)[0])
dx_jit = np.asarray(jax.vjp(jax.jit(gather), x)[1](g)[0])
g = np.asarray(g)

bad = np.where((np.abs(dx_no_jit - g) > 1e-3).any(axis=(1, 2)))[0]
print(f"eager vs expected : max_abs_err = {np.abs(dx_no_jit - g).max():.4f}   -> {'OK' if len(bad)==0 else 'WRONG'}")
print(f"jit   vs expected : max_abs_err = {np.abs(dx_jit - g).max():.2e}   -> {'OK' if np.allclose(dx_jit, g, atol=1e-3) else 'WRONG'}")

# one element dx[:, 0, 0] across the batch (each should equal g[:, 0, 0]).
half = B // 2
print(f"\ndx[row, 0, 0] vs expected across the {B} batch rows:")
for r in [0, 1, half - 1, half, half + 1, B - 1]:
    mark = "  <- ZEROED (should be nonzero!)" if dx_no_jit[r, 0, 0] == 0.0 and abs(g[r, 0, 0]) > 1e-6 else ""
    print(f"  row {r:3d}:  eager = {dx_no_jit[r, 0, 0]: .5f}   expected = {g[r, 0, 0]: .5f}{mark}")
if len(bad):
    print(f"\neager zeroed {len(bad)} of {B} batch rows (rows {bad[0]}..{B-1}); expected all nonzero.")

```

**Expected:** both eager and jit return g unchanged.

**Actual:** (jax 0.10.2 / libtpu 0.0.42.1): dx_no_jit matches g for batch rows 0–127 but is exactly 0.0 for rows 128–255; dx_jit is correct.

**Note:** The trigger for this depends on the dimensions and is non-monotonic in all three:
  * B: ok for B <= 192, fails for B >= 256.
  * D (scatter target): fails at multiples of 8 (8, 16); 9..15 pass.
  * K: fails at 127/128/129 and 256; passes at 120 and 192. 127/129 corrupt all rows; 128 only the second half.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.10.2
jaxlib: 0.10.2
numpy:  2.4.4
python: 3.12.13 (main, Jun  2 2026, 22:27:15) [Clang 22.1.3 ]
device info: TPU v4-4, 4 local devices
process_count: 1
platform: uname_result(system='Linux', node='t1v-n-a2361288-w-0', release='5.8.0-1035-gcp', version='#37~20.04.1-Ubuntu SMP Thu Jun 17 16:04:29 UTC 2021', machine='x86_64')
```

**Comment by jakevdp** [comment]
Thanks for the report – seems like an XLA:TPU bug.

**Comment by jakevdp** [comment]
Internal report: b/533050992

**Comment by patrick-toulme** [comment]
Fixed internally. Thanks for reporting. was bug in compiler
